### ISSUE TITLE
Merge dropping comments

### ISSUE THREAD

**Issue by dirkmueller** [mikefarah/yq#677]
**Describe the bug**

The upgrading from v3 notes are not clear on how to restore the v3 behavior of merging two yaml trees with preserving the comments of the original yaml. 

The usecase I have is that I want to merge a values override list of yaml into the main values.yml, preserving any comments along the way. 

with yq v3, this worked using this command line:

```
$ yq m -x values.yaml my-values.yaml
```

Now, after a few minutes of reading, I figured that this is in v4 syntax this command

```
$ yq eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' values.yaml my-values.yaml
```

with the difference, that the resulting output has no comment (neither head nor line comments) preserved from the original values.yaml. 

is there a way to restore this behavior? the documentation is pretty unclear. 

Version of yq: 4.4.1
Operating system: linux
Installed via: binary release

**Input Yaml**
values.yaml:
```yaml
expose:
  # Set the way how to expose the service. Set the type as "ingress",
  # "clusterIP", "nodePort" or "loadBalancer" and fill the information
  # in the corresponding section
  type: ingress
  tls:
    # Enable the tls or not. Note: if the type is "ingress" and the tls
    # is disabled, the port must be included in the command when pull/push
    # images. Refer to https://github.com/goharbor/harbor/issues/5291
    # for the detail.
    enabled: true
```

my-values.yaml:
```yaml
expose:
  tls:
    auto:
      rotateCert: false
```

**Command**
The command you ran:
```
yq eval-all 'select(fileIndex == 0) * select(fileIndex == 1)' values.yaml my-values.yaml
```

**Actual behavior**

```yaml
expose:
  type: ingress
  tls:
    enabled: true
    auto:
      rotateCert: false
```

**Expected behavior**

```yaml
expose:
  # Set the way how to expose the service. Set the type as "ingress",
  # "clusterIP", "nodePort" or "loadBalancer" and fill the information
  # in the corresponding section
  type: ingress
  tls:
    # Enable the tls or not. Note: if the type is "ingress" and the tls
    # is disabled, the port must be included in the command when pull/push
    # images. Refer to https://github.com/goharbor/harbor/issues/5291
    # for the detail.
    enabled: true
    auto:
        rotateCert: false
```

**Comment by mikefarah** [comment]
Looks like this is a bug - you shouldn't need to do anything to keep comments.

Looks like lineComments are preserved (e.g. `a: cat # line comment`, but the block comments are not :(

Will look into it

**Comment by mikefarah** [comment]
The underlying yaml parser associates the block level comments with the map key, not the value - and `yq` is not merging map keys, which is why it's getting dropped, and also why the object level anchors are getting dropped.

**Comment by mikefarah** [comment]
Fixed in 4.5.1
